src/tree/tree.rs

```diff
@@ -131,100 +108,21 @@ where
     }
 
     /// Return the current root/state of the tree.
-    pub fn root(&self) -> Result<BlsScalar, Error<S::Error>> {
+    pub fn root(&self) -> Result<BlsScalar, CanonError> {
         self.branch(0).map(|b| b.unwrap_or_default().root().clone())
     }
 
-    /// Iterates over the tree, provided its annotation implements [`PoseidonWalkableAnnotation`]
-    pub fn iter_walk<D: Clone>(
+    /// Provides an iterator over the leaves of the tree from a provided starting point.
+    /// To iterate the entire tree, simply provide `0` as `start`.
+    pub fn iter_walk(
```

Comment:
This change looks like a regression. We had a more powerful iterator that could iter from generic criteria and we reduced it for an incremented index. Maybe we just want to create a different method?

Reply:
You're right!. It has been added back in 98e9d2c